dgl_sparse/src/sparse_matrix.cc

```diff
@@ -122,6 +124,57 @@ c10::intrusive_ptr<SparseMatrix> SparseMatrix::FromDiag(
   return SparseMatrix::FromDiagPointer(diag, value, shape);
 }
 
+c10::intrusive_ptr<SparseMatrix> SparseMatrix::IndexSelect(
+    int64_t dim, torch::Tensor ids) {
+  auto id_array = TorchTensorToDGLArray(ids);
+  bool rowwise = dim == 0;
+  SparseFormat sparse_format;
+  if (rowwise) {
+    sparse_format = SparseFormat::kCSR;
+  } else {
+    sparse_format = SparseFormat::kCSC;
+  }
+  auto csr = rowwise ? this->CSRPtr() : this->CSCPtr();
+  auto slice_csr = dgl::aten::CSRSliceRows(CSRToOldDGLCSR(csr), id_array);
+  auto slice_value =
+      this->value().index_select(0, DGLArrayToTorchTensor(slice_csr.data));
+  // Drop the index array to prevent potential construct error of COO format.
```

Comment:
Can you be more specific here, where does the error come from?

Reply:
OK, and the reason is here: https://github.com/dmlc/dgl/pull/6208#discussion_r1314173011.